src/librustc_trans/trans/intrinsic.rs

```diff
@@ -1716,3 +1578,39 @@ fn generic_simd_intrinsic<'blk, 'tcx, 'a>
     }
     bcx.sess().span_bug(call_info.span, "unknown SIMD intrinsic");
 }
+
+// Returns the width of an int TypeVariant, and if it's signed or not
+// Returns None if the type is not an integer
```

Comment:
I just checked, you can call `t.bit_width()` below, but that doesn't help with `TyIs` and `TyUs`.
I'm surprised, but I guess your solution is good (you could put this somewhere more general, like a `Ty` method, though).